### PR TITLE
[WIP] Warn users about accessing internally-intended functions

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

Since many of our methods are `public`, it seems some end-users are utilizing functions that were intended for internal-only use. This PR adds deprecation notices to those methods to warn end-users to not utilize these functions. After enough time has passed with the deprecation notices, many of these functions should be switched to `internal`, `protected` or `private` access levels to make them inaccessible to end-users.

- Delete unnecessary IResource interface
- Mark non end-user functions as obsolete, change access level in the future

# Testing

All unit tests pass

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: EasyPost/Resource.cs [hunk 1]
```diff
@@ -64,11 +67,11 @@ public void Merge(object source)
 
             switch (value)
             {
-                case IResource resource:
+                case Resource resource:
```
Comment (Justintime50) [inline EasyPost/Resource.cs:70]: What's with the `IResource` to `Resource` change?

Comment (nwithan8) [reply EasyPost/Resource.cs:70]: IResource was an interface, I'm assuming added for reusability way-back-when, but it's not being used anywhere except here, and since it's an interface, it's interfering with the ability to change the access limit on some functions.

TL;DR old unnecessary code that was removed.

**Review by Justintime50 — APPROVED** [review]

**Review by jchen293 — APPROVED** [review]
